File: 351004/Beliaev/restapp/discussion/src/main/java/com/example/discussion/mapper/NoteMapper.java
package com.example.discussion.mapper;

import com.example.discussion.dto.request.NoteRequestTo;
import com.example.discussion.dto.response.NoteResponseTo;
import com.example.discussion.model.Note;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public interface NoteMapper {
    Note toEntity(NoteRequestTo request);
    NoteResponseTo toResponse(Note entity);
    void updateEntityFromDto(NoteRequestTo request, @MappingTarget Note entity);
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/mapper/NoteMapper.kt
package com.example.distcomp.mapper

import com.example.distcomp.dto.request.NoteRequestTo
import com.example.distcomp.dto.response.NoteResponseTo
import com.example.distcomp.model.Note
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface NoteMapper {
    fun toEntity(request: NoteRequestTo): Note
    fun toResponse(entity: Note): NoteResponseTo
}
